index.bs

```diff
@@ -9824,6 +9824,71 @@ Methods</h5>
 		</div>
 </dl>
 
+<h5 id="AudioWorkletProcessor-instantiation">
+The instantiation of AudioWorkletProcessor</h5>
+
+At the end of the {{AudioWorkletNode}} construction,
+A [=struct=] named
+<dfn>processor construction data</dfn>
+will be prepared for the cross-thread transfer. This
+<a spec="infra" lt="struct">struct</a> contains the following
+<a spec="infra" for="struct" lt="item">items</a>:
+
+- <dfn for="processor construction data">name</dfn> which is a {{DOMString}}
+	that is to be looked up in the
+	<a>node name to processor constructor map</a>.
+- <dfn for="processor construction data">node</dfn> which is a reference to
+	the {{AudioWorkletNode}} created.
+- <dfn for="processor construction data">options</dfn> which is a serialized
+	{{AudioWorkletNodeOptions}} given to the {{AudioWorkletNode}}'s
+	{{AudioWorkletNode()|constructor}}.
+- <dfn for="processor construction data">port</dfn> which is a serialized
+	{{MessagePort}} paired with the {{AudioWorkletNode}}'s
+	{{AudioWorkletNode/port}}.
+
+Upon the arrival of the transferred data on the {{AudioWorkletGlobalScope}},
+the <a>rendering thread</a> will invoke the algorithm below:
+
+	<div id="invoking-processor-constructor" algorithm="invoking processor constructor">
+		1. Let <var>constructionData</var> be the
+			[=processor construction data=] transferred from the
+			[=control thread=].
+
+		1. Let <var>processorName</var> be <var>constructionData</var>'s
+			[=processor construction data/name=].
+
+		1. Let <var>nodeReference</var> be <var>constructionData</var>'s
+			[=processor construction data/node=].
+
+		1. Let <var>serializedPort</var> be <var>constructionData</var>'s
+			[=processor construction data/port=].
+		
+		1. Let <var>serializedOptions</var> be
+			<var>constructionData</var>'s
+			[=processor construction data/options=].
+		
+		1. Let <var>deserializedPort</var> be the result of
+			[$StructuredDeserialize$](<var>serializedPort</var>,
+			the current Realm).
```

Comment:
What current Realm?  There is no current Realm here.

Reply:
Okay, then how do we deserialize the transferred data without a `realm`? I took the text for granted when I got some help from @domenic. Perhaps the `realm` should be given to the algorithm from the call site?

Reply:
Also this algorithm is performed in AudioWorkletGlobalScope so there is a defined "current realm", I think.
- https://tc39.es/ecma262/#current-realm
- https://www.w3.org/TR/worklets-1/#creating-a-workletglobalscope

Reply:
You want the realm of the AudioWorkletGlobalScope, but it's not the current realm at that point.  In order for it to be the current realm, you'd have to have a running execution context, which can only happen if something pushes it on the execution context stack, etc.  None of that is happening here.

What you probably want to say is that deserialization is passed the realm of the AudioWorkletGlobalScope involved.  It doesn't look like ES really defines getting the realm of a global; I guess you can say you use the realm whose  [[GlobalObject]]  is the AudioWorkletGlobalScope that corresponds to the BaseAudioContext of the AudioWorkletNode that we're in the middle of creating....

Reply:
I guess maybe https://html.spec.whatwg.org/multipage/webappapis.html#concept-global-object-realm offers a shortcut, so you can just talk about taking the realm of the AudioWorkletGlobalScope etc.

Reply:
Thanks for the pointer, but this is really out of scope of this PR. I file a new issue and work on a new PR for it.